pkg/kv/kvserver/batcheval/cmd_delete_range.go

```diff
@@ -150,9 +159,25 @@ func DeleteRange(
 			// written. Thus, setting NumKeys would just reduce the limit available to
 			// the next range for no good reason.
 		}
-		// Return result is always empty, since the reply is populated into the
-		// passed in resp pointer.
-		return result.Result{}, err
+
+		var res result.Result
+		// If GCHint was provided, then we need to check if this request meets range
+		// gc criteria of removing all data.
+		if args.GcRangeHint != nil {
+			if args.Key.Equal(desc.StartKey.AsRawKey()) && args.EndKey.Equal(desc.EndKey.AsRawKey()) {
```

Comment:
You'll want to do this earlier, in the  `MVCCDeleteRangeUsingTombstone` branch. This branch here with `MVCCPredicateDeleteRange` may only remove parts of the range data.